client/tests/integration/non_mintable.rs

```diff
@@ -8,30 +8,27 @@ use iroha_data_model::{metadata::UnlimitedMetadata, prelude::*};
 use test_network::*;
 
 #[test]
-fn non_mintable_asset_can_be_minted_once_but_not_twice() -> Result<()> {
+fn non_mintable_asset_can_be_registered_but_cannot_be_minted() -> Result<()> {
     let (_rt, _peer, mut test_client) = <PeerBuilder>::new().start_with_runtime();
     wait_for_genesis_committed(&vec![test_client.clone()], 0);
 
     // Given
     let account_id = AccountId::from_str("alice@wonderland").expect("Valid");
     let asset_definition_id = AssetDefinitionId::from_str("xor#wonderland").expect("Valid");
     let create_asset =
-        RegisterBox::new(AssetDefinition::quantity(asset_definition_id.clone()).mintable_once());
+        RegisterBox::new(AssetDefinition::quantity(asset_definition_id.clone()).not_mintable());
```

Comment:
I don't see how `mintable_once` is problematic, in this context: these assets can really be minted once.

Reply:
After this PR they become actually non-mintable, only registrable. See my other [comment](https://github.com/hyperledger/iroha/pull/2333#discussion_r900331226)